src/components/Message/Message.tsx

```diff
@@ -575,6 +636,83 @@ const MessageWithContext = <
       }, [] as Reactions)
     : [];
 
+  const handleToggleReaction = async (reactionType: string) => {
+    const messageId = message.id;
+    const ownReaction = !!reactions.find(
+      (reaction) => reaction.own && reaction.type === reactionType,
+    );
+
+    // Change reaction in local state, make API call in background, revert to old message if fails
+    try {
+      if (channel && messageId) {
+        if (ownReaction) {
+          await channel.deleteReaction(messageId, reactionType);
+        } else {
+          await channel.sendReaction(
+            messageId,
+            {
+              type: reactionType,
+            } as Reaction<Re, Us>,
+            { enforce_unique: enforceUniqueReaction },
+          );
+        }
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const handleResendMessage = () => {
+    const messageWithourReservedFields = removeReservedFields(message);
+
+    return retrySendMessage(
+      messageWithourReservedFields as MessageResponse<At, Ch, Co, Me, Re, Us>,
+    );
+  };
+
+  const handleReplyMessage = () => {
+    setQuotedMessageState(message);
+  };
+
+  const isMuted = (client.mutedUsers || []).some(
```

Comment:
Can we remove the repeat of this const later using this one?

Reply:
Done!!